int16 mode
int16 current
int16 velocity
int16 position